compiler/layouter.nim

```diff
@@ -227,8 +242,11 @@ proc wrNewline(em: var Emitter; kind = ltCrucialNewline) =
   em.col = 0
 
 proc newlineWasSplitting*(em: var Emitter) =
+  # sometimes (but not always) there are spaces at position ^2
   if em.kinds.len >= 3 and em.kinds[^3] == ltCrucialNewline:
     em.kinds[^3] = ltSplittingNewline
+  elif em.kinds.len >= 2 and em.kinds[^2] == ltCrucialNewline:
```

Comment:
Is this *only* required by the line 474 ``newlineWasSplitting(em)``? Because if so, please don't call ``newlineWasSplitting(em)`` in line 474 and inline this check instead.

Reply:
It is also called from `parser.nim`:

```nim
template newlineWasSplitting(p: var TParser) =
  when defined(nimpretty):
    layouter.newlineWasSplitting(p.em)
```

Reply:
I know but the existing usages worked fine with ``^3``.